### ISSUE TITLE
prjresults doesn't support repo/arch options

### ISSUE THREAD

**Issue by Jellyfrog** [openSUSE/osc#341]
While it claims support:
```
osc help pr
prjresults (pr): Shows project-wide build results

Usage:
    osc prjresults (inside working copy)
    osc prjresults PROJECT

Options:
    -h, --help          show this help message and exit
    --show-excluded     show packages that are excluded in all repos, also
                        hide repos that have only excluded packages
    -V, --vertical      list packages vertically instead horizontally
    -r REPO, --repo=REPO
                        show results only for specified repo(s)
    -a ARCH, --arch=ARCH
                        show results only for specified architecture(s)
...
```

It doesn't use these flags: 
```bash
$ osc -d -A opensuse results YaST:Head -r openSUSE_Tumbleweed -a x86_64
GET https://api.opensuse.org/build/YaST:Head/_result
...
```
Since its never parsed here; https://github.com/openSUSE/osc/blob/08504a1509761a82d57fddfb10e824830c55c96f/osc/core.py#L5548-L5551

Compared to `osc results` which parses the flags:
https://github.com/openSUSE/osc/blob/08504a1509761a82d57fddfb10e824830c55c96f/osc/core.py#L5527-L5545

**Comment by marcus-h** [comment]
On 2017-09-27 14:21:29 +0000, Jellyfrog wrote:
> While it claims support:
> ```
> osc help pr
> prjresults (pr): Shows project-wide build results
> 
> Usage:
>     osc prjresults (inside working copy)
>     osc prjresults PROJECT
> 
> Options:
>     -h, --help          show this help message and exit
>     --show-excluded     show packages that are excluded in all repos, also
>                         hide repos that have only excluded packages
>     -V, --vertical      list packages vertically instead horizontally
>     -r REPO, --repo=REPO
>                         show results only for specified repo(s)
>     -a ARCH, --arch=ARCH
>                         show results only for specified architecture(s)
> ...
> ```
> 
It really supports --repo and --arch;)

> It doesn't use these flags: 
> ```bash
> $ osc -d -A opensuse results YaST:Head -r openSUSE_Tumbleweed -a x86_64
> GET https://api.opensuse.org/build/YaST:Head/_result
> ...
> ```
> Since its never parsed here; https://github.com/openSUSE/osc/blob/08504a1509761a82d57fddfb10e824830c55c96f/osc/core.py#L5548-L5551
> 
It does some client-side filtering in get_prj_results. Probably we
should switch to the server-side filtering, though.

> Compared to `osc results` which parses the flags:
> https://github.com/openSUSE/osc/blob/08504a1509761a82d57fddfb10e824830c55c96f/osc/core.py#L5527-L5545

The command from above does the right, doesn't it? (Although, there
is a small bug, in the way we pass the options from "results" to
"prjresults" (but we do not hit it in this scenario...))

**Comment by Jellyfrog** [comment]
@marcus-h sorry I missed the last of the command: `--xml`

```
osc -d -A opensuse results YaST:Head -r openSUSE_Tumbleweed -a x86_64 --xml | grep "<result project"
GET https://api.opensuse.org/build/YaST:Head/_result
  <result project="YaST:Head" repository="openSUSE_Tumbleweed" arch="i586" code="published" state="published">
  <result project="YaST:Head" repository="openSUSE_Tumbleweed" arch="x86_64" code="published" state="published">
  <result project="YaST:Head" repository="openSUSE_Leap_42.3" arch="x86_64" code="published" state="published">
  <result project="YaST:Head" repository="openSUSE_Leap_42.2" arch="x86_64" code="published" state="published">
  <result project="YaST:Head" repository="openSUSE_Factory" arch="x86_64" code="published" state="published">
  <result project="YaST:Head" repository="openSUSE_Factory" arch="i586" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="x86_64" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="i586" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="ppc" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="ppc64" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="s390x" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="aarch64" code="published" state="published">
  <result project="YaST:Head" repository="images" arch="ppc64le" code="published" state="published">
``` 

as you can see, its not filtered.
When doing it on a package it filters properly:

```
osc -d -A opensuse results YaST:Head/yast2-network -r openSUSE_Tumbleweed -a x86_64 --xml                                                                                                                    ⏎
GET https://api.opensuse.org/build/YaST:Head/_result?multibuild=1&locallink=1&package=yast2-network&repository=openSUSE_Tumbleweed&arch=x86_64
<resultlist state="ee6f2cff2850c794470220cade95a927">
  <result project="YaST:Head" repository="openSUSE_Tumbleweed" arch="x86_64" code="published" state="published">
    <status package="yast2-network" code="succeeded" />
  </result>
</resultlist>
```

**Comment by marcus-h** [comment]
On 2017-09-28 07:21:40 +0000, Jellyfrog wrote:
> @marcus-h sorry I missed the last of the command: `--xml`
> 
Ah ok, now it makes sense:) Fixed in commit f52cf77.
Thanks for the report!
